### DISCUSSION TITLE
Paths with symlinks inside do not compare as equal

### DISCUSSION THREAD

**Discussion by frozolotl** [helix-editor/helix#11386] [General]
### Summary

When opening a path that does not contain a symlink and then one that does, two buffers are opened, even if the referenced files are the same.

This sounds like a rare and insignificant issue, but it isn't. Clangd's go to definition implementation seems to always yield a canonicalized path. Combine that with the home directory being a symlink (which is the default on Fedora Kinoite) and you will get a horrible DX, where a `g d` press opens a buffer different from the one already opened. This in turn causes edits in one buffer to not appear in the other.

Using a bisect I could determine that the bug was exacerbated in 8444f52. I don't think it is the original cause though; it just caused me to notice the issue.

### Reproduction Steps

1. `mkdir actual`
2. `ln -s actual/ symlink`
3. `echo something > actual/test.txt`
4. `hx --vsplit actual/test.txt symlink/test.txt`

At this point, you will have two buffers open that reference the same file. Edits in one buffer will not appear in the other one.

### Helix log

I don't think it helps much, but here it is anyway:
https://paste.gg/p/anonymous/fe9473d1c89d46f6991f72863130f594

### Platform

Linux, Fedora Kinoite

### Terminal Emulator

Wezterm 20240203-110809-5046fc22

### Installation Method

Built from source (nix profile install)

### Helix Version

3fcf168c

**Comment by pascalkuthe** [comment]
This is a bug in clangd
 Thai behavior is intentional you can have multiple swmanticaldifferent files linked to the same location. Our behavior matches other editors like vim.
